src/main/java/teammates/sqllogic/core/CoursesLogic.java

```diff
@@ -60,6 +69,72 @@ public Course getCourse(String courseId) {
         return coursesDb.getCourse(courseId);
     }
 
+    /**
+     * Returns a list of {@link Course} for all courses a given student is enrolled in.
+     *
+     * @param googleId The Google ID of the student
+     */
+    public List<Course> getCoursesForStudentAccount(String googleId) {
+        List<Student> students = usersLogic.getAllStudentsByGoogleId(googleId);
+
+        List<String> courseIds = students.stream()
+                .filter(student -> !getCourse(student.getCourseId()).isCourseDeleted())
+                .map(student -> student.getCourseId())
+                .collect(Collectors.toList());
+
+        return coursesDb.getCourses(courseIds);
+    }
+
+    /**
+     * Returns a list of {@link CourseAttributes} for all courses for a given list of instructors
+     * except for courses in Recycle Bin.
+     */
+    public List<Course> getCoursesForInstructors(List<Instructor> instructorsList) {
+        assert instructorsList != null;
+
+        List<String> courseIdList = instructorsList.stream()
+                .filter(instructor -> !coursesDb.getCourse(instructor.getCourseId()).isCourseDeleted())
+                .map(Instructor::getCourseId)
+                .collect(Collectors.toList());
+
+        List<Course> courseList = coursesDb.getCourses(courseIdList);
```

Comment:
`coursesDB::getCourses` shouldn't be necessary.

```suggestion
        List<Course> courses = instructorsList.stream()
                .map(Instructor::getCourse)
                .filter(course -> !course.isCourseDeleted())
                .collect(Collectors.toList());
```